### PR TITLE
adding common test setup and teardown to install and uninstall rook.

### LINKED ISSUE #771: Integration Test refactor  - Set up test fixtures so rook can be installed and cleaned up per suite.
Need to refactor this after refactor efforts from PR 761.

### PR DESCRIPTION
**Pull request by dangula**
adding test fixtures to install and uninstall rook per test suite. 
fix #771  and #702

### CODE REVIEW

File: tests/README.md [hunk 1]
```diff
@@ -1,32 +1,40 @@
 # Rook Test Framework
 
-The Rook Test Framework is a Go project that that uses a Docker container with Kubernetes
-tools and Docker pre-installed to quickly create a three node Kubernetes cluster using Docker in
-Docker (DinD) techniques, installs Rook and then runs automation tests to verify its functionality.
+The Rook Test Framework is used to run end to end and integration tests on rook. The framework depends kubernetes
```
Comment (bassam) [inline tests/README.md]: "depends on"

File: tests/README.md [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
-         ```e2e/scripts/smoke_test.sh```
+###Set up
+#### Install Kubernetes
+Users can choose any kubernetes flavor of their choice.  The Test framework only depends on kubectl being configured. 
+The framework also provides scripts to install Kubernetes. Run [kubeadm-dind.sh](/tests/scripts/kubeadm-dind.sh) to setup
+a three node kubernetes cluster using kubeadm(using docker in docker. each node is a docker container running kubernetes) 
+or run [minikube.sh](/tests/scripts/minikube.sh)to set up a single node kubernetes using kubeadm. 
+  
+  Both the start up scrits are designed to start kubrnetes, make sure rbd is working and copy latest rook and toolbox images
+  to the kubernetes containers.
```
Comment (bassam) [inline tests/README.md]: you should note that minikube does no currently include cephfs.ko. @kokhang is going add it

File: tests/baseSetup.go [hunk 3]
```diff
@@ -20,40 +20,44 @@ import (
 	"fmt"
```
Comment (bassam) [inline tests/baseSetup.go]: I think this file should move a level down. its only applicable for file, block, and and object storage suites which assume rook is already installed. it should not run, for example, when we are testing the operator itself.

maybe get rid of the this file here and just have each of the test suites that need rook installed call rookHelper directly?

File: tests/baseSetup.go [hunk 4]
```diff
@@ -20,40 +20,44 @@ import (
...
 	if err != nil {
 		panic(fmt.Errorf("Cannot get platform %v", err))
 	}
+	Platform = platform
+	rook, err := installer.NewK8sRookhelper()
 
-	k8sVersion, _ := enums.GetK8sVersionFromString(Env.K8sVersion)
-
-	err, rookInfra := rook_test_infra.GetRookTestInfraManager(rookPlatform, true, k8sVersion)
+	skipRookInstall := strings.EqualFold(Env.SkipInstallRook, "true")
```
Comment (bassam) [inline tests/baseSetup.go]: why is this needed?

File: tests/baseSetup.go [hunk 5]
```diff
@@ -20,40 +20,44 @@ import (
...
 	Platform enums.RookPlatformType
 )
 
 //One init function per package - initializes Rook infra and installs rook(if needed based on flags)
 func init() {
 	Env = objects.NewManifest()
 	var err error
-
-	rookPlatform, err := enums.GetRookPlatFormTypeFromString(Env.Platform)
+	platform, err := enums.GetRookPlatFormTypeFromString(Env.Platform)
```
Comment (bassam) [inline tests/baseSetup.go]: is this still needed? I thought we're going to remove the platform abstraction and go with specific test suites for each platform?

File: tests/framework/installer/install_helper.go [hunk 6]
```diff
@@ -156,12 +162,68 @@ func (h *RookHelper) InstallRook() (err error) {
 	return nil
 }
 
-func NewRookHelper() (*RookHelper, error) {
+func (h *InstallHelper) UninstallRookFromK8s() {
 
-	transportClient := transport.CreateNewk8sTransportClient()
+	k8sHelp := utils.CreatK8sHelper()
+	var err error
+	_, err = k8sHelp.ResourceOperation("delete", path.Join(getPodSpecPath(), rookOperatorFileName))
```
Comment (bassam) [inline tests/framework/installer/install_helper.go]: is k8sHelp just a wrapper on top of kubectl? If so I think it we be great if it where change to

```
func kubectl(args ...)
```

and it logged and panic inside the function so that this function would look like:

```
kubectl("delete",rookOperatorFileName")
kubectl("delete","rook","cluster","rook")
```

File: tests/framework/installer/install_helper.go [hunk 7]
```diff
@@ -156,12 +162,68 @@ func (h *RookHelper) InstallRook() (err error) {
...
+	_, err = k8sHelp.DeleteResource([]string{"secret", "rook-rook-user"})
+	if err != nil {
+		panic(err)
+	}
+	_, err = k8sHelp.DeleteResource([]string{"namespace", "rook"})
+	if err != nil {
+		panic(err)
+	}
+
+	args := []string{"-n", "rook", "pods", "-l", "app=rook-ceph-mon"}
```
Comment (bassam) [inline tests/framework/installer/install_helper.go]: this would be good to generalize. `func waitUntilPodIsGone(podName string)`

File: tests/README.md (hunk 2 above)
Comment (dangula) [inline tests/README.md]: I have updated the minikube.sh script. we are using @kokhang 's minikube iso image to start minikube.

File: tests/README.md (hunk 1 above)
Comment (dangula) [inline tests/README.md]: done.

File: tests/README.md (hunk 2 above)
Comment (kokhang) [inline tests/README.md]: I have submitted a PR to minikube. Once merged, it should be available in a couple of weeks on the next build of minikube. In the mean time, you can use this minikube iso i built https://s3-us-west-2.amazonaws.com/minikube-cephfs/minikube.iso

**Comment by bassam** [comment]
Can you squash please

**Comment by dangula** [comment]
squashed commits.
